contract_comm/blockchain_parameters/blockchain_parameters.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Celo Authors
+// This file is part of the celo library.
+//
+// The celo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The celo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the celo library. If not, see <http://www.gnu.org/licenses/>.
+
+package blockchain_parameters
+
+import (
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/contract_comm"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+const (
+	blockchainParametersABIString = `[{
+			"constant": true,
+			"inputs": [],
+			"name": "getMinimumClientVersion",
+			"outputs": [
+			  {
+				"name": "major",
+				"type": "uint256"
+			  },
+			  {
+				"name": "minor",
+				"type": "uint256"
+			  },
+			  {
+				"name": "patch",
+				"type": "uint256"
+			  }
+			],
+			"payable": false,
+			"stateMutability": "view",
+			"type": "function"
+	}]`
+)
+
+const defaultGasAmount = params.ContractCommGas
+
+var blockchainParametersABI abi.ABI
+
+func init() {
+	var err error
+	blockchainParametersABI, err = abi.JSON(strings.NewReader(blockchainParametersABIString))
+	if err != nil {
+		log.Error("Error reading ABI for BlockchainParameters", "err", err)
```

Comment:
probably fatal here, since this happens on startup and the erros means a developer bug